build.zig

```diff
@@ -546,3 +581,52 @@ fn dotnet_client(
         build_step.dependOn(&lib.step);
     }
 }
+
+fn node_client(
+    b: *std.build.Builder,
+    mode: Mode,
+    dependencies: []const *std.build.Step,
+    options: *std.build.OptionsStep,
+    tracer_backend: config.TracerBackend,
+) void {
+    const build_step = b.step("node_client", "Build Node client shared library");
+
+    for (dependencies) |dependency| {
+        build_step.dependOn(dependency);
+    }
+
+    // Zig cross-targets:
+    const platforms = .{
+        "x86_64-linux-gnu",
+        "x86_64-linux-musl",
+        "x86_64-macos",
+        "aarch64-linux-gnu",
+        "aarch64-linux-musl",
+        "aarch64-macos",
+        // "x86_64-windows", # x86_64-windows-gnu is broken for now
+    };
+
+    inline for (platforms) |platform| {
+        const cross_target = CrossTarget.parse(.{ .arch_os_abi = platform, .cpu_features = "baseline" }) catch unreachable;
+
+        const lib = b.addSharedLibrary("tb_nodeclient", "src/clients/node/src/node.zig", .unversioned);
+        lib.setMainPkgPath("src");
+        lib.setOutputDir("src/clients/node/dist/bin/" ++ platform);
+        lib.addSystemIncludeDir("/usr/include");
```

Comment:
This feels very non-hermetic (eg, I don't have `/usr/include` on my machine). Why do we need this? Is this to make `node_headers.h` work? I _think_ zig ships some include files, I wonder if we can use those?

Reply:
Yeah, it's to make `node_api.h` work (shipped by Node). I do agree hardcoding it in like this isn't ideal (perhaps there's a better way of getting `zig build` to include the default system include folders?)

My thought is when running `zig build node_client` you don't expect it to be hermetic but rather to build against the standard system headers like a normal package would. In CI, we make this hermetic by explicitly providing a fixed version of Node & headers.

Reply:
@matklad I've reworked this a bit. Since we have the `node-api-headers` package as a dev-dependency for future Windows support, the build can just reference that and not rely on any system paths or includes.